app/src/common/UsersDataRequests.js

```diff
@@ -0,0 +1,38 @@
+import $ from 'jquery'
+import axios from 'axios';
+import endPointsList from './endPointsList'
+
+class UsersDataRequests {
+
+    static signUpReq(newUser) {
+        return axios.post(`${endPointsList.serverUrl}${endPointsList.saveNewUser}`, newUser)
+            .then((res) => {
```

Comment:
I'd recommend passing callback function through the parameters to the function. So UI logic (show, hide, alert and so on in the future) will store in the Component which invokes this function. And this function will be clean and do only it's job - make the requests.
Do the same in the similar places in this class.